src/api/Nhs.Appointments.Api/Models/TypeFixDocumentFilter.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Abstractions;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nhs.Appointments.Api.Models;
+
+public class TypeFixDocumentFilter(IEnumerable<Type> typesToFix, TimeProvider time) : IDocumentFilter
+{    
+    public void Apply(IHttpRequestDataObject req, OpenApiDocument document)
+    {
+        foreach (var schema in document.Components.Schemas)
+        {
+            var type = typesToFix.SingleOrDefault(t => t.Name.ToLower() == schema.Key.ToLower());
+            if (type == null)
+                continue;
+
+            if (schema.Value.Type == "object")
+            {
+                var updatedSchema = new OpenApiSchema
+                {
+                    Type = "object",
+                    Properties = new Dictionary<string, OpenApiSchema>()
+                };
+
+                var clrProperties = type.GetProperties();
+
+                foreach (var prop in schema.Value.Properties)
+                {
+                    var targetProperty = clrProperties.Single(p => p.Name.ToLower() == prop.Key.ToLower());
+                    updatedSchema.Properties.Add(prop.Key, FixupSchema(targetProperty.PropertyType, prop.Value));                    
+                }
+
+                document.Components.Schemas[schema.Key] = updatedSchema;
+            }
+        }
+    }
+
+    private OpenApiSchema FixupSchema(Type type, OpenApiSchema originalSchema) 
+    {        
+        var underlyingType = Nullable.GetUnderlyingType(type) ?? type;
+
+        if(underlyingType.IsEnum)
+        {
+            return new OpenApiSchema
+            {
+                Type = "string",
+                Enum = type.GetEnumNames().Select(n => new OpenApiString(n)).Cast<IOpenApiAny>().ToList(),
+                Default = new OpenApiString(type.GetEnumNames().Last())
```

Comment:
I'm absolutely fine with it, but out of curiousity why have we chosen the default to be the last here? Just surprised me since when newing up an enum 0 would be the default

Reply:
In some paradigms the default for an enum will (and arguably should) be undefined. This is so that they don't have a meaningful value if they are not set. An example of this is that QueryType the default (0) value is actually NotSet. To prevent these from showing I wanted to use a value other than the default. I chose last to prevent having to test for weird edge cases where there was only one value in the enum